src/Spork/SharedMem.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of Spork, an OpenSky project.
+ *
+ * (c) OpenSky Project Inc
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Spork;
+
+use Spork\Exception\ProcessControlException;
+
+/**
+ * Sends messages between processes.
+ */
+class SharedMem
+{
+    private $pid;
+    private $ppid;
+    private $signal;
+    private $isParent;
+
+    /**
+     * Constructor.
+     *
+     * @param integer $pid    The child process id or null if this is the child
+     * @param integer $signal The signal to send after writing to shared memory
+     */
+    public function __construct($pid = null, $signal = null)
+    {
+        if (null === $pid) {
+            // child
+            $pid   = posix_getpid();
+            $ppid  = posix_getppid();
+            $this->isParent = false;
+        } else {
+            // parent
+            $ppid  = null;
+            $this->isParent = true;
+        }
+
+        $this->pid  = $pid;
+        $this->ppid = $ppid;
+        $this->signal = $signal;
+    }
+
+    /**
+     * Reads all messages from shared memory.
+     *
+     * @return array An array of messages
+     */
+    public function receiveMessages()
```

Comment:
Thanks Oleg. I took a look again at this code, and realized that this method wasn't actually working as advertised. I think that I fixed the logic now, so take another look when you can.